src/api.cc

```diff
@@ -170,6 +170,36 @@ class LiteInstall : public InstallContext {
     return DownloadResult{DownloadResult::Status::Ok, ""};
   }
 
+  void QueueEvent(std::string ecu_serial, SecondaryEvent event, std::string details) override {
+    Uptane::EcuSerial serial(ecu_serial);
+    std::unique_ptr<ReportEvent> e;
+    if (event == InstallContext::SecondaryEvent::DownloadStarted) {
+      e = std::make_unique<EcuDownloadStartedReport>(serial, target_->correlation_id());
```

Comment:
I see why `InstallContext` was introduced, now. It looks like too much of a hassle as for the single ID  :).

Reply:
I don't think I really get it. What makes it easier than just calling API functions directly?

Reply:
I think calling the "download" and "install" API methods is the easiest and most flexible way.
Our backend binds "download" and "install" related events together by means of "correlation-ID" and then it's used as a kind of "update-ID" by `fioctl` utility.

```
# Show the details of an update:
fioctl devices updates <device> <update-id>
```

```
mike@msul:~$ fioctl devices updates msul-local-9834 
ID                                        TIME                  VERSION  TARGET
--                                        ----                  -------  ------
996-2d0cb8ff-8437-49a1-8c90-b1065f2d5a0b  2021-09-14T12:23:38Z  996      intel-corei7-64-lmp-996
996-a96bbd9f-17be-4a59-a645-b71d3b37a731  2021-09-14T11:37:34Z  996      intel-corei7-64-lmp-996
```

```
mike@msul:~$ fioctl devices updates msul-local-9834 996-2d0cb8ff-8437-49a1-8c90-b1065f2d5a0b
2021-09-14T12:23:38+00:00 : EcuDownloadStarted(intel-corei7-64-lmp-996)
 Details:
 | Updating from unknown to intel-corei7-64-lmp-996
2021-09-14T12:23:45+00:00 : EcuDownloadCompleted(intel-corei7-64-lmp-996) -> Succeed
2021-09-14T12:23:45+00:00 : EcuInstallationStarted(intel-corei7-64-lmp-996)
2021-09-14T12:23:47+00:00 : EcuInstallationApplied(intel-corei7-64-lmp-996)
 Details:
 | Application successful, need reboot
 | # Apps installed:
 | hub.foundries.io/msul-dev01/app-05@sha256:32a0309bbc1ab28a81fcd012a53f7e2075633b711e75b38c8e2aecaa361aaab6
 | # Apps running:
 | app-05: hub.foundries.io/msul-dev01/app-05@sha256:32a0309bbc1ab28a81fcd012a53f7e2075633b711e75b38c8e2aecaa361aaab6; state: Created
 | 	python-www: 2ed96ec5a5d5615302fe318c9d5433386299bb90cc6d0ebce258979838c26df3; image: hub.foundries.io/msul-dev01/app-05@sha256:5f80d09611ee4f0ba07e7318f9c2970a2d8d0ca84e187e99eae971189a207a71; state: running; status: Up 50 minutes
 | 
2021-09-14T12:23:52+00:00 : EcuInstallationCompleted(intel-corei7-64-lmp-996) -> Succeed
 Details:
 | Successfully booted on new version
 | # Apps running:
 | app-05: hub.foundries.io/msul-dev01/app-05@sha256:32a0309bbc1ab28a81fcd012a53f7e2075633b711e75b38c8e2aecaa361aaab6; state: Started
 | 	python-www: 2ed96ec5a5d5615302fe318c9d5433386299bb90cc6d0ebce258979838c26df3; image: hub.foundries.io/msul-dev01/app-05@sha256:5f80d09611ee4f0ba07e7318f9c2970a2d8d0ca84e187e99eae971189a207a71; state: running; status: Up 50 minutes
 | 

```